### PR TITLE
feat(auth): enable emulator support on desktop

### PR DESCRIPTION
**Pull request by cynthiajoan**
[replace this line]: # (Describe your changes in detail.)
***

[replace this line]: # (Describe your testing in detail.)
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: auth/src/android/auth_android.cc [hunk 1]
```diff
@@ -199,6 +199,8 @@ void CheckEmulator(AuthData* auth_data) {
     return;
   }
 
+  LogDebug("Using Auth Emulator for testing.");
```
Comment (jonsimantov) [inline auth/src/android/auth_android.cc:202]: Want to make this a full-on LogInfo?

File: auth/src/desktop/rpcs/auth_request.cc [hunk 2]
```diff
@@ -88,6 +88,7 @@ std::string AuthRequest::GetUrl() {
   } else {
     std::string url(kHttp);
     url += emulator_url;
+    url += "/";
```
Comment (jonsimantov) [inline auth/src/desktop/rpcs/auth_request.cc:91]: Just making sure this still works when *not* using emulator?

File: release_build_files/readme.md [hunk 3]
```diff
@@ -627,6 +627,12 @@ workflow use only during the development of your app, not for publicly shipping
 code.
 
 ## Release Notes
+### Next Release
+-   Changes
+    - Auth: Add Firebase Emulator support. Can set environment attribute
```
Comment (jonsimantov) [inline release_build_files/readme.md:632]: Suggested rephrase:

Auth: Add Firebase Auth Emulator support. Set the environment variable USE_AUTH_EMULATOR=yes (and optionally AUTH_EMULATOR_PORT, default 9099) to connect to the local Firebase Auth Emulator.

**Review by jonsimantov — APPROVED** [review]

File: auth/src/desktop/rpcs/auth_request.cc (hunk 2 above)
Comment (cynthiajoan) [reply auth/src/desktop/rpcs/auth_request.cc:91]: yup verified both scenario.

File: release_build_files/readme.md (hunk 3 above)
Comment (cynthiajoan) [reply release_build_files/readme.md]: done

File: auth/src/android/auth_android.cc (hunk 1 above)
Comment (cynthiajoan) [reply auth/src/android/auth_android.cc]: done

**Pull request merged**